app/src/main/java/de/westnordost/streetcomplete/util/LatLonRaster.kt

```diff
@@ -48,14 +48,14 @@ class LatLonRaster(bounds: BoundingBox, private val cellSize: Double) {
         val startY = max(0, min(latitudeToCellY(bounds.minLatitude), rasterHeight - 1))
         val endX = max(0, min(longitudeToCellX(bounds.maxLongitude), rasterWidth - 1))
         val endY = max(0, min(latitudeToCellY(bounds.maxLatitude), rasterHeight - 1))
-        val result = MultiIterable<LatLon>()
+        var result = sequenceOf<LatLon>()
         for (y in startY..endY) {
             for (x in startX..endX) {
                 val list = raster[y * rasterWidth + x]
-                if (list != null) result.add(list)
+                if (list != null) result += list
             }
         }
-        return result
+        return result.asIterable()
```

Comment:
Another one to avoid mutation
```kotlin
        return sequence<LatLon> {
            for (y in startY..endY) {
                for (x in startX..endX) {
                    raster[y * rasterWidth + x]?.let{ yieldAll(it) }
                }
            }
        }.asIterable()
```

Reply:
Fair enough. Not harder readable because you need to know about sequences in either case.

Reply:
Though I personally find this
```kotlin
val list = raster[y * rasterWidth + x]
if (list != null) yieldAll(list)
```
a little easier to read. Certainly a matter of taste.

Reply:
Begrudgingly, yeah, this probably makes sense, for consistency with the rest of the app. Or a compromise :laughing: 
```kotlin
val list = raster[y * rasterWidth + x]
list?.let { yieldAll(it) }
```